core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -501,7 +501,7 @@ public void deleteClosedIndex(String reason, IndexMetaData metaData, ClusterStat
      * Deletes the index store trying to acquire all shards locks for this index.
      * This method will delete the metadata for the index even if the actual shards can't be locked.
      */
-    public void deleteIndexStore(String reason, IndexMetaData metaData, ClusterState clusterState) throws IOException {
+    public void deleteIndexStore(String reason, IndexMetaData metaData, ClusterState clusterState, boolean closed) throws IOException {
```

Comment:
I actually think rather than passing in the flag for the closed, we could infer from the cluster state whether the index is closed, however, passing a flag in is less complexity since we have a separate `deleteClosedIndex` so we already know.

Anyone have thoughts about this?